### PR TITLE
GH-41968: [Java] Implement TransferPair functionality for BinaryView

### PR DESCRIPTION
**Pull request by vibhatha**
### Rationale for this change

This PR contains the transferPair functionality for BinaryView vectors. 

### What changes are included in this PR?

This includes the addition of transferPair functionality in `ViewVarCharBinaryVector` and corresponding test cases. 

### Are these changes tested?
Yes

### Are there any user-facing changes?

No
* GitHub Issue: #41968

### CODE REVIEW

File: java/vector/src/test/java/org/apache/arrow/vector/TestSplitAndTransfer.java [hunk 1]
```diff
@@ -282,35 +303,52 @@ public void testMemoryConstrainedTransfer() {
...
-        tp.splitAndTransfer(start, length);
-        newViewVarCharVector.clear();
-      }
+    final TransferPair tp = vector.getTransferPair(allocator);
+    final BaseVariableWidthViewVector newVector;
+    if (vector instanceof ViewVarCharVector) {
+      newVector = (ViewVarCharVector) tp.getTo();
+    } else {
+      newVector = (ViewVarBinaryVector) tp.getTo();
+    }
```
Comment (lidavidm) [inline java/vector/src/test/java/org/apache/arrow/vector/TestSplitAndTransfer.java:329]: Hmm, why do we need the branch/cast here? Why can't we just have

```
newVector = (BaseVariableWidthViewVector) tp.getTo();
```

**Review by lidavidm — APPROVED** [review]

File: java/vector/src/test/java/org/apache/arrow/vector/TestSplitAndTransfer.java (hunk 1 above)
Comment (vibhatha) [reply java/vector/src/test/java/org/apache/arrow/vector/TestSplitAndTransfer.java:329]: Oh! On it.

Comment (vibhatha) [reply java/vector/src/test/java/org/apache/arrow/vector/TestSplitAndTransfer.java:329]: There are multiple branches in other test cases, let me fix them shortly.

Comment (vibhatha) [reply java/vector/src/test/java/org/apache/arrow/vector/TestSplitAndTransfer.java]: Just fixed them.

**Comment by vibhatha** [comment]
@lidavidm Updated the PR.

**Review by lidavidm — APPROVED** [review]

**Pull request merged**